WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -296,5 +300,44 @@ public TransactionFactory(Network network, KeyManager keyManager, ICoinsView coi
 			var spendsUnconfirmed = spentCoins.Any(c => !c.Confirmed);
 			return new BuildTransactionResult(new SmartTransaction(tx, Height.Unknown), psbt, spendsUnconfirmed, sign, fee, feePc, outerWalletOutputs, innerWalletOutputs, spentCoins);
 		}
+
+		private PSBT TryNegotiatePayjoin(IPayjoinClient payjoinClient, TransactionBuilder builder, PSBT psbt)
+		{
+			try
+			{
+				psbt = payjoinClient.RequestPayjoin(psbt,
+					KeyManager.ExtPubKey,
+					new RootedKeyPath(KeyManager.MasterFingerprint.Value, KeyManager.DefaultAccountKeyPath),
+					CancellationToken.None).GetAwaiter().GetResult();
+				builder.SignPSBT(psbt);
+			}
```

Comment:
I think it could be good to add some log entry here, something like "Payjoin payment with `https://demo.whatever.org` was negociated successfully.